### PR TITLE
Moved all RealmTests to new guidelines.

### PR DESCRIPTION
**Pull request by cmelchior**
All RealmTests have been upgraded to the new guidelines. Especially Handler tests have all moved to RunTestInLooperThread. Hopefully this will also make them less flaky.

@realm/java

### CODE REVIEW

File: realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java [hunk 1]
```diff
@@ -34,11 +35,18 @@
 /**
  * Rule that runs the test inside a worker looper thread. This Rule is responsible
  * of creating a temp directory containing a Realm instance then delete it, once the test finishes.
+ *
+ * All Realms used in a method method annotated with {@code @RunTestInLooperThread } should use
```
Comment (zaki50) [inline realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java]: `method method`

**Comment by beeender** [comment]
Test like `getInstance_referenceCounting_doubleClose` has multiple lines, more than of them could throw a `IllegalStateException`. I suggest we use `ExpectedException` in such cases to make the test more strict.

**Comment by beeender** [comment]
https://github.com/realm/realm-java/pull/2074/files Take a look of the `findbugs-filter.xml`, some findbugs errors need to be filtered.

**Comment by cmelchior** [comment]
@beeender I like that as a guideline. `@Test(expected` for oneliner tests, `throws.expect` for multiple lines. I'll change my tests.

File: realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java [hunk 2]
```diff
@@ -34,11 +35,18 @@
 /**
  * Rule that runs the test inside a worker looper thread. This Rule is responsible
```
Comment (kneth) [inline realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java]: "This Rule" -> "This rule"

File: realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java [hunk 3]
```diff
@@ -34,11 +35,18 @@
...
  * of creating a temp directory containing a Realm instance then delete it, once the test finishes.
+ *
+ * All Realms used in a method method annotated with {@code @RunTestInLooperThread } should use
+ * {@link RunInLooperThread#createConfiguration()} and friends to create their configurations. Failing to do so can
+ * result in the test failing because the Realm could not be deleted (Reason is that {@link TestRealmConfigurationFactory}
+ * and this class do not agree in which order to delete all open Realms.
```
Comment (kneth) [inline realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java]: "class do not" -> "class does not"

File: realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java [hunk 4]
```diff
@@ -112,4 +121,11 @@ public void run() {
     public void testComplete() {
         signalTestCompleted.countDown();
     }
+
+    /**
+     * Post a runnable to this worker threads looper.
```
Comment (kneth) [inline realm/realm-library/src/androidTest/java/io/realm/rule/RunInLooperThread.java]: "Post" -> "Posts"

**Comment by kneth** [comment]
:-+1: with minor comments of documentation.

**Comment by cmelchior** [comment]
All comments fixed @realm/java

**Comment by beeender** [comment]
:+1:

**Comment by emanuelez** [comment]
Nice! :+1:

**Pull request merged**